netconf.c

```diff
@@ -897,12 +903,36 @@ get_full_tree ()
 }
 
 static gboolean
-remove_null_data (GNode *node, gpointer data)
+process_subtree_query_leaves (GNode *node, gpointer data)
 {
-    if (node->data == NULL)
+    GNode *qnode;
+    q_param *qparam = (q_param *) data;
+    int depth = 0;
+
+    /* Subtree queries have null or value terminated trees */
+    node = node->parent;
+
+    if (node)
     {
-        g_node_unlink (node);
-        g_node_destroy (node);
+        qnode = node;
+        while (qnode->parent)
+        {
+            qnode = qnode->parent;
+            depth++;
+        }
+        depth++;
+
+        if (g_strcmp0 (APTERYX_NAME(node), "*") == 0)
+        {
+            qparam->deepest_leaf = node;
+            qparam->depth = depth;
+            return true;
+        }
+        else if (depth > qparam->depth)
+        {
+            qparam->deepest_leaf = node;
+            qparam->depth = depth;
```

Comment:
this just falls through to return false, what does that mean?

Reply:
This is a g_node_traverse callback routine. If the callback returns false the next traverse node is fetched. If true is returned the tree traversal stops. In this case we want to see all leaves to find the longest path unless a node is found with a "*" as the data. In that case we stop the traversal (line 929).

Reply:
OK